.github/workflows/roxygen.yaml

```diff
@@ -124,6 +148,22 @@ jobs:
           run-system-dependencies: ${{ inputs.install-system-dependencies }}
           direction: ${{ inputs.sd-direction }}
 
+      - name: Setup R dependencies 🎦
+        if: >-
+          env.deps_installation_method == 'setup-r-dependencies'
+        uses: insightsengineering/setup-r-dependencies@v1
+        with:
+          lookup-refs: ${{ inputs.lookup-refs }}
+          github-token: ${{ steps.github-token.outputs.token }}
+          repository-path: ${{ github.event.repository.name }}
+
+      # setup-r-dependencies makes modifications to DESCRIPTION which
+      # should not be committed during Roxygen auto-update.
+      - name: Restore DESCRIPTION ⚙️
+        run: git restore DESCRIPTION
+        working-directory: ${{ github.event.repository.name }}
+        shell: bash
```

Comment:
I think that this should be handled in the setup-r-deps itself. It's would be better for that step to cleanup its own temp stuff
In addition, `git restore` might not be the best way to do this as this includes also all the previous (maybe relevant) changes.

Reply:
I think it's a good idea, I can make this update, but I'm not sure what method other than `git restore` to use. Do we expect any other steps in the workflows (before `setup-r-dependencies`) to modify `DESCRIPTION`? If yes, then how can we distinguish the changes made to it by `setup-r-dependencies` (which we want to discard) and the ones made by the previous steps (which we want to keep)?
One idea I have is to checkout the repository in an additional path/directory only for the purpose of `setup-r-dependencies` modifications to `DESCRIPTION` and resulting installation of dependencies, and this additional path/directory can later be discarded.

Reply:
I was thinking about something more _light_. Since we are modifying just one `DESCRIPTION` file, at the beginning, we can store the content of it (somewhere, somehow) and then at the very end _restore_ it.
It can be done by copying file into temp dir or copy into the current path as `.DESCRIPTION_ORIGINAL` and then overwride (and remove). We can also echo into temp env var or anything like that.

Reply:
True 😅🤦 let me add this.

Reply:
I think this approach is incompatible with the rhub.yaml workflow where `r-hub/actions/setup-deps` relies on modifications to `DESCRIPTION` made by `insightsengineering/setup-r-dependencies`. Maybe we could have an input to `insightsengineering/setup-r-dependencies` deciding whether to restore or not restore the original `DESCRIPTION`?

Reply:
Ahh yes I forgot about that. Then yes let's introduce a parameter with a default to restore and in rhub we will set it to not restore